pkgs/os-specific/linux/minimal-bootstrap/gzip/default.nix

```diff
@@ -0,0 +1,58 @@
+{ lib
+, fetchurl
+, bash
+, tinycc
+, gnumake
+, gnused
+, gnugrep
+}:
+let
+  pname = "gzip";
+  version = "1.2.4";
+
+  src = fetchurl {
+    url = "mirror://gnu/gzip/gzip-${version}.tar.gz";
+    sha256 = "0ryr5b00qz3xcdcv03qwjdfji8pasp0007ay3ppmk71wl8c1i90w";
+  };
+in
+bash.runCommand "${pname}-${version}" {
+  inherit pname version;
+
+  nativeBuildInputs = [
+    tinycc.compiler
+    gnumake
+    gnused
+    gnugrep
+  ];
+
+  passthru.tests.get-version = result:
```

Comment:
since these now take a `result` we need to feed that in for the test now right?

Reply:
It's automatically fed in from `derivationWithMeta` :D
https://github.com/NixOS/nixpkgs/blob/4aa442cab08366eeec3945a46ffa8e67449649b8/pkgs/os-specific/linux/minimal-bootstrap/utils.nix#L26

Reply:
ahh. so does everything in `passthru.tests` need to be a function that takes self/result?

Reply:
Just for these bootstrapping things. `mkDerivation` does it differently with `finalAttrs: { ... }`.